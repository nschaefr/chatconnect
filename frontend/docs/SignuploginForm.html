<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: SignuploginForm</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: SignuploginForm</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>SignuploginForm<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="SignuploginForm"><span class="type-signature"></span>new SignuploginForm<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Component SignupLoginForm
A Component which can switch between a
Sign Up and Login form</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_signuplogin-page_signuplogin-form.jsx.html">components/signuplogin-page/signuplogin-form.jsx</a>, <a href="components_signuplogin-page_signuplogin-form.jsx.html#line12">line 12</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~submitHandler"><span class="type-signature">(async, inner) </span>submitHandler<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Function to check if the password and username are valid</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_signuplogin-page_signuplogin-form.jsx.html">components/signuplogin-page/signuplogin-form.jsx</a>, <a href="components_signuplogin-page_signuplogin-form.jsx.html#line33">line 33</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="~validatePassword"><span class="type-signature">(inner) </span>validatePassword<span class="signature">(password)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Validates a Password</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>password</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Password that user typed in</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_signuplogin-page_signuplogin-form.jsx.html">components/signuplogin-page/signuplogin-form.jsx</a>, <a href="components_signuplogin-page_signuplogin-form.jsx.html#line60">line 60</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="~validateUsername"><span class="type-signature">(inner) </span>validateUsername<span class="signature">(username)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Validates a username</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>username</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Username that user typed in</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_signuplogin-page_signuplogin-form.jsx.html">components/signuplogin-page/signuplogin-form.jsx</a>, <a href="components_signuplogin-page_signuplogin-form.jsx.html#line73">line 73</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Avatar.html">Avatar</a></li><li><a href="Chat.html">Chat</a></li><li><a href="Contact.html">Contact</a></li><li><a href="Routes.html">Routes</a></li><li><a href="SignuploginForm.html">SignuploginForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#App">App</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.2</a> on Thu Sep 07 2023 17:10:02 GMT+0200 (Central European Summer Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>