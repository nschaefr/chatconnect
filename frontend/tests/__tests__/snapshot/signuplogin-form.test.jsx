import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import SignuploginForm from "../../../src/components/signuplogin-page/signuplogin-form";

describe("signuploginform snapshot-tests", () => {
  it("rendering signuploginform", () => {
    const { container } = render(<SignuploginForm />);

    expect(container).toMatchSnapshot();
  });

  it("rendering signupform after click on 'create an account'", () => {
    const { container } = render(<SignuploginForm />);
    fireEvent.click(screen.getByTestId("create-account"));

    expect(container).toMatchSnapshot();
  });

  it("rendering loginform after click on 'login now'", () => {
    const { container } = render(<SignuploginForm />);
    fireEvent.click(screen.getByTestId("create-account"));
    fireEvent.click(screen.getByTestId("login-now"));

    expect(container).toMatchSnapshot();
  });
});
