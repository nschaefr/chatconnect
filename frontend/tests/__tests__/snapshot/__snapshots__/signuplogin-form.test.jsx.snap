// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupLoginForm Snapshot-Tests Rendering LoginForm after click on 'Login now' 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="headerDiv"
    >
      <img
        src="[object Object]"
        width="40px"
      />
      <p
        class="header"
      >
        chatconnect
      </p>
    </div>
    <p
      class="slogan"
    >
      Sign In
    </p>
    <form>
      <input
        class="input"
        data-testid="username-input"
        placeholder="Username"
        type="name"
      />
      <br />
      <input
        class="pwd"
        data-testid="password-input"
        placeholder="Password"
        type="password"
      />
    </form>
    <div
      class="decision"
    >
      <div>
        <p
          class="text"
        >
          You haven't signed up yet?
        </p>
        <p
          class="btn"
          data-testid="create-account"
        >
          Create an account
        </p>
      </div>
      <div
        class="btn2"
      >
        <p
          class="text2"
          data-testid="submit-button"
        >
          Log In
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`SignupLoginForm Snapshot-Tests Rendering SignupForm after click on 'Create an Account' 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="headerDiv"
    >
      <img
        src="[object Object]"
        width="40px"
      />
      <p
        class="header"
      >
        chatconnect
      </p>
    </div>
    <p
      class="slogan"
    >
      Sign Up
    </p>
    <form>
      <input
        class="input"
        data-testid="username-input"
        placeholder="Username"
        type="name"
      />
      <br />
      <input
        class="pwd"
        data-testid="password-input"
        placeholder="Password"
        type="password"
      />
    </form>
    <div
      class="decision"
    >
      <div>
        <p
          class="text"
        >
          Already have an account?
        </p>
        <p
          class="btn"
          data-testid="login-now"
        >
          Login now
        </p>
      </div>
      <div
        class="btn2"
      >
        <p
          class="text2"
          data-testid="submit-button"
        >
          Sign Up
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`SignupLoginForm Snapshot-Tests Rendering SignupLoginForm 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="headerDiv"
    >
      <img
        src="[object Object]"
        width="40px"
      />
      <p
        class="header"
      >
        chatconnect
      </p>
    </div>
    <p
      class="slogan"
    >
      Sign In
    </p>
    <form>
      <input
        class="input"
        data-testid="username-input"
        placeholder="Username"
        type="name"
      />
      <br />
      <input
        class="pwd"
        data-testid="password-input"
        placeholder="Password"
        type="password"
      />
    </form>
    <div
      class="decision"
    >
      <div>
        <p
          class="text"
        >
          You haven't signed up yet?
        </p>
        <p
          class="btn"
          data-testid="create-account"
        >
          Create an account
        </p>
      </div>
      <div
        class="btn2"
      >
        <p
          class="text2"
          data-testid="submit-button"
        >
          Log In
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`signuploginform snapshot-tests rendering loginform after click on 'login now' 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="headerDiv"
    >
      <img
        src="[object Object]"
        width="40px"
      />
      <p
        class="header"
      >
        chatconnect
      </p>
    </div>
    <p
      class="slogan"
    >
      Sign In
    </p>
    <form>
      <input
        class="input"
        data-testid="username-input"
        placeholder="Username"
        type="name"
      />
      <br />
      <input
        class="pwd"
        data-testid="password-input"
        placeholder="Password"
        type="password"
      />
    </form>
    <div
      class="decision"
    >
      <div>
        <p
          class="text"
        >
          You haven't signed up yet?
        </p>
        <p
          class="btn"
          data-testid="create-account"
        >
          Create an account
        </p>
      </div>
      <div
        class="btn2"
      >
        <p
          class="text2"
          data-testid="submit-button"
        >
          Log In
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`signuploginform snapshot-tests rendering signupform after click on 'create an account' 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="headerDiv"
    >
      <img
        src="[object Object]"
        width="40px"
      />
      <p
        class="header"
      >
        chatconnect
      </p>
    </div>
    <p
      class="slogan"
    >
      Sign Up
    </p>
    <form>
      <input
        class="input"
        data-testid="username-input"
        placeholder="Username"
        type="name"
      />
      <br />
      <input
        class="pwd"
        data-testid="password-input"
        placeholder="Password"
        type="password"
      />
    </form>
    <div
      class="decision"
    >
      <div>
        <p
          class="text"
        >
          Already have an account?
        </p>
        <p
          class="btn"
          data-testid="login-now"
        >
          Login now
        </p>
      </div>
      <div
        class="btn2"
      >
        <p
          class="text2"
          data-testid="submit-button"
        >
          Sign Up
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`signuploginform snapshot-tests rendering signuploginform 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="headerDiv"
    >
      <img
        src="[object Object]"
        width="40px"
      />
      <p
        class="header"
      >
        chatconnect
      </p>
    </div>
    <p
      class="slogan"
    >
      Sign In
    </p>
    <form>
      <input
        class="input"
        data-testid="username-input"
        placeholder="Username"
        type="name"
      />
      <br />
      <input
        class="pwd"
        data-testid="password-input"
        placeholder="Password"
        type="password"
      />
    </form>
    <div
      class="decision"
    >
      <div>
        <p
          class="text"
        >
          You haven't signed up yet?
        </p>
        <p
          class="btn"
          data-testid="create-account"
        >
          Create an account
        </p>
      </div>
      <div
        class="btn2"
      >
        <p
          class="text2"
          data-testid="submit-button"
        >
          Log In
        </p>
      </div>
    </div>
  </div>
</div>
`;
